lib/tasks/addon-install.js

```diff
@@ -34,22 +34,23 @@ module.exports = Task.extend({
 
     ui.startProgress(chalk.green('Installing addon package'), chalk.green('.'));
 
+    var willSave = this.project.isEmberCLIAddon() && commandOptions.save;
     return npmInstall.run({
       packages: packageNames,
-      'save-dev': true,
+      'save-dev': !willSave,
       'save-exact': true
     }).then(function() {
       return this.project.reloadAddons();
     }.bind(this)).then(function() {
-      return this.installBlueprint(blueprintInstall, packageNames, blueprintOptions);
+      return this.installBlueprint(blueprintInstall, packageNames);
```

Comment:
Code and tests seem to indicate there is no breakage here.

But I'm an untrained eye when it comes to the source code of ember-cli, so thanks for asking for another perspective.